Packages/Status/Sources/Status/Row/Subviews/StatusRowTextView.swift

```diff
@@ -6,17 +6,47 @@ struct StatusRowTextView: View {
   @EnvironmentObject private var theme: Theme
 
   let status: AnyStatus
-  let viewModel: StatusRowViewModel
-
+  @ObservedObject var viewModel: StatusRowViewModel
+  
   var body: some View {
-    HStack {
-      EmojiTextApp(status.content, emojis: status.emojis, language: status.language)
-        .font(.scaledBody)
-        .emojiSize(Font.scaledBodyPointSize)
-        .environment(\.openURL, OpenURLAction { url in
-          viewModel.routerPath.handleStatus(status: status, url: url)
-        })
-      Spacer()
+    VStack {
+      HStack {
+        EmojiTextApp(status.content, emojis: status.emojis, language: status.language, lineLimit: viewModel.collapsedLines)
+          .font(.scaledBody)
+          .emojiSize(Font.scaledBodyPointSize)
+          .environment(\.openURL, OpenURLAction { url in
+            viewModel.routerPath.handleStatus(status: status, url: url)
+          })
+        Spacer()
+      }
+      makeCollapseButton()
+    }
+  }
+  
+  @ViewBuilder
+  func makeCollapseButton() -> some View {
+    if viewModel.showCollapseButton && viewModel.isCollapsed {
+      HStack(alignment: .top) {
+        Text("status.show-full-post")
+          .font(.system(.subheadline, weight: .bold))
+          .foregroundColor(.secondary)
+        Spacer()
+        Button {
+          withAnimation {
+            viewModel.isCollapsed.toggle()
+          }
+        } label: {
+          Image(systemName: "chevron.down")
+        }
+        .buttonStyle(.bordered)
+        .accessibility(label: Text("status.show-full-post"))
+        .accessibilityHidden(true)
+      }
```

Comment:
Without this, one can tap between the words and the button on the far right and it will take them to the details page instead of expanding the post.
```suggestion
      }
      .contentShape(Rectangle())
```

Reply:
thx, will adjust that - and fix it in `StatusRowSpoilerView` as well where I copied my code from ;-)